Adding Success Screen + microchanges

JimBratsos opened the pull request:
Adding Success Screen folder, which contains the JS file, the SCSS and the components folder which is a folder with all the SVG used in the process. Also tweaked some values in SCSS for landing page without any significant results ( I forgot to revert them back to normal, but no side effects come from it ). Finally I fixed a typo in landing page. Thanks

JimBratsos committed:
Adding forgot-password page, by following the signup page template. Tweaked the values in SCSS file of forgotpassword to make the box smaller, and added comments in forgotpassword

JimBratsos committed:
Merge remote-tracking branch 'upstream/development' into newchanges

JimBratsos committed:
Adding Success screen

ArnasDickus committed:
Merge branch 'development' into newchanges

ArnasDickus commented:
@JimBratsos There was a conflict in app.js regarding, comments, so I fix it for you.

ArnasDickus commented on src/components/LandingPage/sections/SectionOne/SectionOne.js:
```diff
@@ -36,7 +36,7 @@ const SectionOne = ({ scrollClick }) => (
         <ParallaxLayer speed={0.1} offset={0.57}>
             <p className={`${classes.contentWrapper} ${classes.tagline}`}>
-                An app that converts your favoriet music pieces into visual
+                An app that converts your favorite music pieces into visual
```
Thanks for fixing spelling mistake.

ArnasDickus commented on src/components/LandingPage/LandingPage.module.scss:
```diff
@@ -9,7 +9,7 @@ $x-large: 1200px;
   left: 0;
   width: 100vw;
-  height: 100vh;
+  height: 500vh;
```
vh is based on %. So 100vh is equal 100% of viewport height. There's no reason to put it at 500vh [Documentation](https://www.sitepoint.com/css-viewport-units-quick-start/).

ArnasDickus commented on src/components/Success/Success.module.scss:
```diff
@@ -0,0 +1,105 @@
+    .avatarProfile{
+        background: url('./Particles/avatar-profile.svg');
+        position: absolute;
+        width: 40px;
```
Could you use rem or em instead of px, for accessibility?

ArnasDickus commented on src/components/Success/Success.js:
```diff
@@ -0,0 +1,28 @@
+                <div className={classes.avatarProfile}></div>
+                <div className={classes.rectangle}>
+                    <div className={classes.guitarist}></div>
+                    <div className={classes.successText}></div>
```
I don't think you should be using SVG instead of text. Although the only argument I have is that Google robot can't read text on SVG.  So if you want to defend your choice. I'm all ears.

ArnasDickus commented on src/components/Success/Success.module.scss:
```diff
@@ -0,0 +1,105 @@
+            top: 450px;
+            
+            font-family: Biryani;
+            font-style: normal;
```
font-style and font-weight are normal by default. So you need to remove it. Also, font-family is already added to global CSS, so you can remove it here.

ArnasDickus commented on src/components/Success/Success.module.scss:
```diff
@@ -0,0 +1,105 @@
+                height: 60px;
+                left: 130px;
+                top: 380px;
+            }
```
For performance, reason try to avoid nesting more when 3 levels.

JimBratsos replied:
As I checked the nesting is exactly 3 levels if I am not wrong. If it does need fixing, please let me now

ArnasDickus replied:
Yes, you right, on that one.

ArnasDickus commented on src/components/Success/Success.module.scss:
```diff
@@ -0,0 +1,105 @@
+.successComponent{
+    background: #000000;
```
The idea is this would be a popup that appears on top, so it doesn't need to have black background.

ArnasDickus reviewed (changes requested):
Hello, thanks for adding a PR.

1) There are some formatting issues. Try running `npm run format`
2) Also, try to use rem or em for accessibility instead of px. At least for sizes that are bigger than 10px.

JimBratsos committed:
Fixing 500 vh to 100 vh

JimBratsos committed:
Adding Success text

JimBratsos committed:
Update Success.module.scss

JimBratsos committed:
Changes in style

Fixing px to em, and also the Success text from SVG to normal text

JimBratsos reviewed (commented):
Fixed the mistakes, awaiting reply to the nesting comment. Waiting for approval

JimBratsos reviewed (commented):
Fixed the mistakes, awaiting reply to the nesting comment. Waiting for approval

ArnasDickus reviewed (changes requested):
Hello, So you seem to have fixed all my issues. However, now there's a problem with the component. It seems background images repeat. Try adding `background-repeat: no-repeat` on background images.
![background-repeat](https://user-images.githubusercontent.com/38657595/66717235-54572d00-eddf-11e9-9132-abddee4842e8.jpg)

JimBratsos committed:
Fixing repeating images

Adding background-repeat: no-repeat at every background image of my component. Hope that fixes the problem. I couldnt test it because I dont know how to make the bug appear

JimBratsos reviewed (commented):
All done, changes made

ArnasDickus reviewed (changes requested):
Hello so your PR still have problems:
1) The component height is too big.
2) Success text is incorrectly placed.
![text-wrong-too-big](https://user-images.githubusercontent.com/38657595/66719060-a30ec200-edf3-11e9-9986-3b1ae17ecc74.jpg)
[Success screen mockup](https://www.figma.com/file/KN4yf9KRGRLxPBlUy4VrixZS/Visual-Music?node-id=852%3A131)

JimBratsos committed:
Fixing the components

Fixing the components' position and size. Made fully responsive, and placed in the middle of the visualizer ( to imitate the popup )

JimBratsos committed:
Adding Avatar component in the box

Adding Avatar component in the box. I know it's 4 levels nested now, but it couldnt be made otherwise.

JimBratsos reviewed (commented):
Changes made as requested

ArnasDickus commented on src/components/Success/Success.module.scss:
```diff
@@ -0,0 +1,130 @@
+    width: 100vw;
+    height: 100vh;
+    display: flex;
+    align-items: center;
```
align-items: center seems to do nothing

ArnasDickus commented on src/components/Success/Success.module.scss:
```diff
@@ -0,0 +1,130 @@
+    .rectangle{
+        top:0em;
+        display:flex;
+        justify-content: center;
```
These lines seem to have no effect on code.
```
display: flex; 
justify-content: center; 
align-content: center;
```

ArnasDickus reviewed (changes requested):
Hello, so according to the [mockup](https://www.figma.com/file/KN4yf9KRGRLxPBlUy4VrixZS/Visual-Music?node-id=852%3A131). It's in the center. You seem to added on top instead of center. Without scrolling I can't see player bar. Also we don't actually know what avatar profile do, so you can remove it.

Looking at current situation, and considering that this is my 3rd review. I suggest to either start from scratch, try to avoid fixed width and height, or leave task to someone else.

![sucess-screen](https://user-images.githubusercontent.com/38657595/66856212-64ad0a80-ef8d-11e9-9700-2207e977db16.jpg)

On mobile, you seem to created a bug of shorting playerbar. Bug appears only when success component is added.
![mobile-success](https://user-images.githubusercontent.com/38657595/66856293-9625d600-ef8d-11e9-8d4e-e2685cb15931.jpg)

ArnasDickus commented:
Hello, @JimBratsos I just approved [PR #46 Error screen](https://github.com/zero-to-mastery/visual-music/pull/46). It might be useful to look into it.

JimBratsos committed:
Updating Success

Fixing all the errors. Putting the component in middle of the screen. Checked all the bugs. None is present Ill post screenshots in a comment

JimBratsos reviewed (commented):
Changes made, tested in most mobile phones and it's working smoothly. Thanks for sticking with me 
![Desktop](https://user-images.githubusercontent.com/48690015/66861162-287ea780-ef97-11e9-85d6-6b684816ddb0.JPG)
![Mobile View](https://user-images.githubusercontent.com/48690015/66861163-287ea780-ef97-11e9-83bc-86f3fcc7a718.JPG)

JimBratsos committed:
Final changes in the file

Turns out that by removing the height of text, the problems with responsiveness were solved. I tweaked some other values too to make the component look great in all the devices. Tested

ArnasDickus reviewed (changes requested):
Hello [mockup](https://www.figma.com/file/KN4yf9KRGRLxPBlUy4VrixZS/Visual-Music?node-id=852%3A131)

The problem with this PR is:
Problem with mobile responsiveness. Also player bar bug.
![sucess-not-responsive](https://user-images.githubusercontent.com/38657595/67244943-8a418480-f463-11e9-911a-f0f93bae6a6f.jpg)

On desktop not fully center:
![on desktop not fully centered](https://user-images.githubusercontent.com/38657595/67244993-a1807200-f463-11e9-88d7-d9fda8a20ba0.jpg)

My suggestion is just to copy [PR #46 css ](https://github.com/zero-to-mastery/visual-music/pull/46)

JimBratsos commented:
The decision giving basically the same task to two different people created lots of confusion. Also, I do not know how it should be to be responsive in mobile. There is no template, or no information whether if it should cover the whole screen or not. It is supposed to be a popup otherwise it will be too small. Lastly, I cannot copy just the CSS, as my JS file and the error component is different. Again , assigning two people with the same thing, made lots of confusion. I will withdraw from this PR, as I cannot make it after all. Sorry
